bluebrain/documentation/using_installed_software_in_an_ide.md

```diff
@@ -0,0 +1,141 @@
+  `spack` uses.
+  * `spack-configure-args.txt` a file with additional arguments passed to CMake.
+  * a folder `spack-build-*` which is the output of running CMake.
+
+Remark: It might be a good idea to use `spack` environments to reduce the
+number of packages that get listed as dependencies.
+
+## Quick method
+Since CMake has been run during the `spack dev-build` and we know the name of
+the folder, one may simply use that folder, i.e. tell your IDE to use this
+folder instead of what it would use normally.
+
+The IDE is unlikely to be able to recreate this folder, but you can simply
+rerun `spack` when needed.
+
+## Using `CMAKE_PREFIX_PATH`
+Let's treat `spack` as a package installer; and CMake as the tool which tells
+the IDE how to configure itself. For our purposed `spack` works as follows:
+
+  * Every dependency is installed into a unique folder.
+  * If multiple versions, also incompatible ones, are installed, `spack` knows
+  which ones should be used together.
+
+The trick is to convince CMake that it should find the correct version of each
+dependency, this is done by setting through `CMAKE_PREFIX_PATH`.
+
+By inspecting `spack-build-env.txt` we see that it sets the environment variable
+`CMAKE_PREFIX_PATH`, containing the long list of paths where each dependency can
+be found.
+
+### Option 1
+Somehow set the same environment variable in the IDE.
+
+### Option 2
+Convert the environment variable into a command line argument
+`-DCMAKE_PREFIX_PATH=...`. Small caveat, the environment variable is separated
+by colons and in the CLI argument paths must be separated by semi-colons.
+
+There's a script for this:
+
+    spack2cmake.sh SOURCE_DIR
+
+which outputs the required CMake command. Here `SOURCE_DIR` is the directory
+containing the `spack-build-env.txt`.
+
+Most IDEs should allow you to pass additional arguments to CMake on a per
+project basis. Read the output of the previous command carefully and copy-paste
+the required parts to your IDE.
+
+## Using `compile_commands.json`
+It's possible that your IDE isn't happy yet. Maybe it simply can't deal directly
+with `CMakeLists.txt`. So here a second common trick. CMake can generate a file
+called `compile_commands.json`. This is a file that various IDEs/editors can
+parse to figure out things like include paths.
+
+How can we generate this file? Use Option 2 in the previous method to generate
+a CMake command. Then add the flag `-DCMAKE_EXPORT_COMPILE_COMMANDS=1`. The
+complete command will look something like:
+
+    cmake -DCMAKE_PREFIX_PATH=... \
+          -DCMAKE_EXPORT_COMPILE_COMMANDS=1 \
+          FURTHER_FLAGS \
+          -B build
+
+Now point your IDE/editor to the generated file, e.g.
+`build/compile_commands.json`.
+
+# Concrete examples
+Let's take `touchdetector` as an example. The steps are to source `spack`
+and then create a new environment
+```
+spack env create TouchDetector
+spack compiler find && spack external find
```

Comment:
This should be in a separate setup step (and out of the scope of this README), not *after* sourcing the environment. If it is done within the environment, chances are that changes to the config are local to the environment.

Reply:
That's what I've been doing, glad this can be simpler.